src/source/Ice/IceAgent.c

```diff
@@ -107,25 +113,40 @@ STATUS createIceAgent(PCHAR username, PCHAR password, PIceAgentCallbacks pIceAge
                                            (PCHAR) pRtcConfiguration->iceServers[i].username, (PCHAR) pRtcConfiguration->iceServers[i].credential),
                 pIceAgent->iceAgentProfileDiagnostics.iceServerParsingTime[i], "ICE server parsing");
             if (STATUS_SUCCEEDED(retStatus)) {
-                pIceAgent->rtcIceServerDiagnostics[i].port = (INT32) getInt16(pIceAgent->iceServers[i].ipAddress.port);
-                switch (pIceAgent->iceServers[pIceAgent->iceServersCount].transport) {
-                    case KVS_SOCKET_PROTOCOL_UDP:
-                        STRCPY(pIceAgent->rtcIceServerDiagnostics[i].protocol, ICE_TRANSPORT_TYPE_UDP);
-                        break;
-                    case KVS_SOCKET_PROTOCOL_TCP:
-                        STRCPY(pIceAgent->rtcIceServerDiagnostics[i].protocol, ICE_TRANSPORT_TYPE_TCP);
-                        break;
-                    default:
-                        MEMSET(pIceAgent->rtcIceServerDiagnostics[i].protocol, 0, SIZEOF(pIceAgent->rtcIceServerDiagnostics[i].protocol));
+                if (!statsControlEnabled || (statsControlEnabled && pIceAgent->kvsRtcConfiguration.enableIceStats)) {
```

Comment:
The second `statsControlEnabled` is redundant.
```suggestion
                if (!statsControlEnabled || pIceAgent->kvsRtcConfiguration.enableIceStats) {
```